### PR TITLE
Changed KyberFieldElement from a struct to a type alias for an i16.

### PR DESCRIPTION
**Pull request by xvzcf**
Also removed the `FieldElement` interface, and the generic PolynomialRingElement interface.

### CODE REVIEW

File: src/kem/kyber768/ntt.rs [hunk 1]
```diff
@@ -41,21 +42,21 @@ pub(crate) mod kyber_polynomial_ring_element_mod {
                 zeta_i += 1;
 
                 for j in offset..offset + layer {
-                    let t = re[j + layer] * ZETAS[zeta_i];
-                    re[j + layer] = re[j] - t;
-                    re[j] = re[j] + t;
+                    let t = fe_mul(re[j + layer], ZETAS[zeta_i]);
```
Comment (franziskuskiefer) [inline src/kem/kyber768/ntt.rs:45]: For readability, we might want to implement the operators. But let's keep this for now.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Thanks, looks good. Only a few nits.

File: src/kem/kyber768/arithmetic.rs
[the remark this answers is not in the document]
Comment (xvzcf) [reply src/kem/kyber768/arithmetic.rs:1]: Added a TODO to add more comments

**Review by franziskuskiefer — APPROVED** [review]
Thanks :shipit:

**Pull request merged**